Modules/Filtering/ImageStatistics/include/itkStatisticsImageFilter.hxx

```diff
@@ -28,224 +28,47 @@ namespace itk
-StatisticsImageFilter< TInputImage >
-::GetMaximumOutput() const
-{
-  return static_cast< const PixelObjectType * >( this->ProcessObject::GetOutput(2) );
-}
-
-template< typename TInputImage >
-typename StatisticsImageFilter< TInputImage >::RealObjectType *
-StatisticsImageFilter< TInputImage >
-::GetMeanOutput()
-{
-  return static_cast< RealObjectType * >( this->ProcessObject::GetOutput(3) );
-}
-
-template< typename TInputImage >
-const typename StatisticsImageFilter< TInputImage >::RealObjectType *
-StatisticsImageFilter< TInputImage >
-::GetMeanOutput() const
-{
-  return static_cast< const RealObjectType * >( this->ProcessObject::GetOutput(3) );
-}
-
-template< typename TInputImage >
-typename StatisticsImageFilter< TInputImage >::RealObjectType *
-StatisticsImageFilter< TInputImage >
-::GetSigmaOutput()
-{
-  return static_cast< RealObjectType * >( this->ProcessObject::GetOutput(4) );
-}
-
-template< typename TInputImage >
-const typename StatisticsImageFilter< TInputImage >::RealObjectType *
-StatisticsImageFilter< TInputImage >
-::GetSigmaOutput() const
-{
-  return static_cast< const RealObjectType * >( this->ProcessObject::GetOutput(4) );
-}
-
-template< typename TInputImage >
-typename StatisticsImageFilter< TInputImage >::RealObjectType *
-StatisticsImageFilter< TInputImage >
-::GetVarianceOutput()
-{
-  return static_cast< RealObjectType * >( this->ProcessObject::GetOutput(5) );
-}
-
-template< typename TInputImage >
-const typename StatisticsImageFilter< TInputImage >::RealObjectType *
-StatisticsImageFilter< TInputImage >
-::GetVarianceOutput() const
-{
-  return static_cast< const RealObjectType * >( this->ProcessObject::GetOutput(5) );
-}
-
-template< typename TInputImage >
-typename StatisticsImageFilter< TInputImage >::RealObjectType *
-StatisticsImageFilter< TInputImage >
-::GetSumOutput()
-{
-  return static_cast< RealObjectType * >( this->ProcessObject::GetOutput(6) );
-}
-
-template< typename TInputImage >
-const typename StatisticsImageFilter< TInputImage >::RealObjectType *
-StatisticsImageFilter< TInputImage >
-::GetSumOutput() const
-{
-  return static_cast< const RealObjectType * >( this->ProcessObject::GetOutput(6) );
-}
-
-
-template< typename TInputImage >
-typename StatisticsImageFilter< TInputImage >::RealObjectType *
-StatisticsImageFilter< TInputImage >
-::GetSumOfSquaresOutput()
-{
-  return static_cast< RealObjectType * >( this->ProcessObject::GetOutput(7) );
-}
-
-template< typename TInputImage >
-const typename StatisticsImageFilter< TInputImage >::RealObjectType *
-StatisticsImageFilter< TInputImage >
-::GetSumOfSquaresOutput() const
-{
-  return static_cast< const RealObjectType * >( this->ProcessObject::GetOutput(7) );
-}
-
-template< typename TInputImage >
-void
-StatisticsImageFilter< TInputImage >
-::GenerateInputRequestedRegion()
-{
-  Superclass::GenerateInputRequestedRegion();
-  if ( this->GetInput() )
+  if ( name == "Mean" ||
```

Comment:
This is ugly, but I guess this is C++, Python is nicer for this. I thought about using a const string results[] = {"Mean", "Sigma", "Variance", "Sum", "SumOfSquares"} and std::find, but probably doesn't look more elegant.